src/Microsoft.ML.Tokenizers/Model/SentencePieceOptions.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Tokenizers
+{
+#pragma warning disable MSML_NoInstanceInitializers
+    /// <summary>
+    /// The type of the SentencePiece model.
+    /// </summary>
+    public enum SentencePieceModelType
+    {
+        /// <summary>
+        /// The model type is not defined.
+        /// </summary>
+        Undefined,
+
+        /// <summary>
+        /// The model type is Byte Pair Encoding (Bpe) model.
+        /// </summary>
+        Bpe,
+
+        /// <summary>
+        /// The model type is Unigram model.
+        /// </summary>
+        Unigram,
+    }
+
+    /// <summary>
+    /// Options for the SentencePiece tokenizer.
```

Comment:
We might want to make note that we don't provide any serialization grantees for this type.  I know that's a general rule for most of our types but it might warrant a callout since we're specifically exposing it as bridge to user serialized data.